File: 알고리즘 with 동빈나/DFS & BFS/Python1/Ex_BFS.py
from collections import deque

# BFS 함수 정의
def bfs(graph, start, visited):
    queue = deque([start])
    # print(type([start]))
    # 현재 노드를 방문 처리
    visited[start] = True
    
    while queue:
        v = queue.popleft()
        print(v, end=' ')

        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True

    
visited = [False] * 9

graph = [
    [],  # 0번 노드... 없으...
    [2, 3, 8],  # 1번 노드와 인접한 노드        ____ 1______
    [1, 7],  # 2번 노드와 인접한 노드         _2    _3_     8
    [1, 4, 5],  # 3번 노드와 인접한 노드    _7     4---5   /  
    [3, 5],  #                           6  \___________/ 
    [3, 4],
    [7],
    [2, 6, 8],
    [1, 7]
]

bfs(graph, 1, visited)

File: 알고리즘 with 동빈나/DFS & BFS/Cpp1/Ex_BFS.cpp
#include <bits/stdc++.h>
using namespace std;


void bfs(vector<vector<int>>& graph, int start, bool(& visited) [9])
{
    queue<int> q;
    q.push(start);
    // 현재 노드를 방문 처리
    visited[start] = true;

    while (!q.empty())
    {
        int v = q.front();
        q.pop();
        cout << v << ' ';
        for (int i = 0; i < graph[v].size(); ++i)
        {
            if (!visited[graph[v][i]])
            {
                q.push(graph[v][i]);
                visited[graph[v][i]] = true;
            }
        }
    }
}

int main_ex_bfs()
{
    vector<vector<int>> graph = {
        {},           // 0번 노드
        {2, 3, 8},    // 1번 노드와 인접한 노드
        {1, 7},       // 2번 노드와 인접한 노드
        {1, 4, 5},    // 3번 노드와 인접한 노드
        {3, 5},       // 4번 노드와 인접한 노드
        {3, 4},       // 5번 노드와 인접한 노드
        {7},          // 6번 노드와 인접한 노드
        {2, 6, 8},    // 7번 노드와 인접한 노드
        {1, 7}        // 8번 노드와 인접한 노드
    };
    // vector<bool> visited(9, false);
    bool visited[9] = { false };
    bfs(graph, 1, visited);

	return EXIT_SUCCESS;
}